allennlp/data/token_indexers/single_id_token_indexer.py

```diff
@@ -8,6 +8,9 @@
 from allennlp.data.token_indexers.token_indexer import TokenIndexer, IndexedTokenList
 
 
+_DEFAULT_VALUE = "THIS IS A REALLY UNLIKELY VALUE THAT HAS TO BE A STRING"
```

Comment:
The way `_DEFAULT_VALUE` is used now, it could just be `None`, with no surprising strings like this one.

Reply:
The reason I did this was to allow `None` to be explicitly passed as a default value, if you wanted that for some reason.  I originally just had `default_value = None` in the constructor, but then you can't actually have `None` as a default value.

Reply:
And then I had `_DEFAULT_VALUE = object()`, but then I got type complaints because `object()` isn't a `str`, which is what the type of `default_value` is annotated as.

Reply:
It's a fair point. I don't like the solution, but I don't have a better idea.